child_compassion/models/child_pictures.py

```diff
@@ -43,21 +43,27 @@ class child_pictures(models.Model):
     #                             FIELDS METHODS                             #
     ##########################################################################
     @api.one
-    def set_pictures(self):
+    def set_pictures(self, typeshot='Headshot'):
         """Get the picture given field_name (headshot or fullshot)"""
         attachment_obj = self.env['ir.attachment']
         attachments = attachment_obj.search([
             ('res_model', '=', self._name),
             ('res_id', '=', self.id)])
 
-        for data in attachments:
-            picturename = data.datas_fname
-            picturename = picturename.split('.')
-            picturename = picturename[0]
-            if picturename == 'Fullshot':
-                self.fullshot = data.datas
-            elif picturename == 'Headshot':
-                self.headshot = data.datas
+        # We sort the images by date of creation from newest to oldest
+        attachments = attachments.sorted(
+            lambda key: key.create_date, reverse=True)

```

Comment:
you can directly add the desired order in the search function at line 51, by adding parameter `order='create_date desc'`